pkg/generators/clientgen/clientgen.go

```diff
@@ -0,0 +1,179 @@
+/*
+Copyright 2022 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package clientgen
+
+import (
+	"path/filepath"
+	"sort"
+	"strings"
+
+	"k8s.io/code-generator/cmd/client-gen/types"
+	"k8s.io/gengo/namer"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-tools/pkg/genall"
+	"sigs.k8s.io/controller-tools/pkg/markers"
+
+	"github.com/kcp-dev/code-generator/pkg/internal/clientgen"
+	"github.com/kcp-dev/code-generator/pkg/parser"
+	"github.com/kcp-dev/code-generator/pkg/util"
+)
+
+type Generator struct {
+	// Name is the name of this client-set, e.g. "kubernetes"
+	Name string `marker:""`
+
+	// HeaderFile specifies the header text (e.g. license) to prepend to generated files.
+	HeaderFile string `marker:",optional"`
+
+	// Year specifies the year to substitute for " YEAR" in the header file.
+	Year string `marker:",optional"`
```

Comment:
We don't want to be including a year in generated code. If it's optional, ok. But if it's required, that's going to be an issue every January.

Reply:
ACK, can likely remove, this was just the pattern in controller-gen